### PR TITLE
Filter out restricted user from Playlist scores

### PR DESCRIPTION
**Pull request by notbakaneko**

### CODE REVIEW

File: app/Http/Controllers/Multiplayer/Rooms/Playlist/ScoresController.php [hunk 1]
```diff
@@ -53,7 +53,12 @@ public function index($roomId, $playlistId)
         $limit = clamp(get_int($params['limit'] ?? null) ?? 50, 1, 50);
         $cursorHelper = PlaylistItemUserHighScore::makeDbCursorHelper($params['sort'] ?? null);
 
-        $highScoresQuery = $playlist->highScores()->whereHas('scoreLink');
+        $highScoresQuery = $playlist
+            ->highScores()
+            ->whereHas('user', function ($userQuery) {
+                $userQuery->default();
+            })
```
Comment (nanaya) [inline app/Http/Controllers/Multiplayer/Rooms/Playlist/ScoresController.php:60]:
```suggestion
            ->whereHas('user', fn ($userQuery) => $userQuery->default())
```

**Review by nanaya — CHANGES_REQUESTED** [review]

File: app/Models/Multiplayer/PlaylistItemUserHighScore.php [hunk 2]
```diff
@@ -130,4 +131,9 @@ public function updateWithScoreLink(ScoreLink $scoreLink): bool
             'total_score' => $score->total_score,
         ])->save();
     }
+
+    public function user(): BelongsTo
```
Comment (nanaya) [inline app/Models/Multiplayer/PlaylistItemUserHighScore.php:135]: oh right, this should be in relation function area?

**Review by nanaya — CHANGES_REQUESTED** [review]

File: app/Models/Multiplayer/PlaylistItemUserHighScore.php (hunk 2 above)
Comment (notbakaneko) [reply app/Models/Multiplayer/PlaylistItemUserHighScore.php:135]: are we still doing that 🤔

Comment (nanaya) [reply app/Models/Multiplayer/PlaylistItemUserHighScore.php]: I think so for scopes, relations and laravel model setters/getters? 🤔

**Review by nanaya — APPROVED** [review]

**Pull request merged**